react/week3/todo-list/src/App.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import './App.css';
+import Header from './Components/Header';
+import SetTimer from './Components/SetTimer';
+import TodoList from "./Components/TodoList";
+import AddTodo from './Components/AddTodo';
+
+function App() {
+    const [todoList, setTodoList] = useState([]);
+
+    useEffect(() => {
+        fetch("https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw")
+            .then(res => res.json())
+            .then(
+                (result) => {
+                    setTodoList(result);
+                },
+            )
+    }, [])
+
+    const addTodo = (userInput, startDate) => {
+        setTodoList([...todoList, {
+            id: todoList.length > 0 ? todoList[todoList.length - 1].id + 1 : 1,
+            description: userInput,
+            deadline: startDate.toISOString().substring(0, 10)
+        }]);
+    }
+
+    const deleteTodo = (id) => {
+        const newTodoList = todoList.filter((task) => task.id !== id);
+        setTodoList(newTodoList);
+    }
+
+    const editTodo = (id, newDescription) => {
+        todoList.map((task) => {
+            if (task.id === id) {
+                task.description = newDescription;
+            }
+            return task;
+        });
```

Comment:
1. The point of the map array method is to create a new array:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

2. States should not be mutated. Any mutation is handled by React which can be achieved through the `setState()` method.

Alternatively you could write your map function this way to reach the desired result:
```
const editedTodoList = todoList.map(task => {
  if(task.id === id){
    return {...task, description: newDescription}
  } else return task
});
```
Here the spread operator creates a copy of the task object ("...task") and after the comma any value can be edited ("description: newDescription")

And then set the state:
`setTodoList(editedTodoList);`

Reply:
Thank you for the time and effort you took to give the feedback. Will modify as you mentioned.